### PR TITLE
Scenes: Implement "Live now" feature

### PR DESCRIPTION
**Pull request by kaydelaney**
First pass at migrating https://github.com/grafana/grafana/pull/37769 to scenes.
Not every nuance of the feature has been migrated, but it's a start.

### CODE REVIEW

**Review by axelavargas — APPROVED** [review]
Awesome work, @kaydelaney 🙌🏾 

I've reviewed the code and it looks good to me. Implementing this as a behavior is a good approach in my opinion.

As you mentioned, there's currently missing logic for the live feature, specifically regarding determining the panel's refresh rate based on both the time range and width.

P.S. I did a small sync locally with the Grafana repository and works as expected.

**Review by ivanortegaalba — COMMENTED** [review]
I like the approach of using a behavior too. We thought to have this only implemented in Dashboard, but I think this is a great addition to the library.

I'd add some tests, and probably documentation as part of the docs when the feature is ready to work as it works in Dashboard.

Great progress on this Kay! Let's add some tests, and LGTM!

File: packages/scenes/src/components/VizPanel/VizPanel.tsx [hunk 1]
```diff
@@ -215,6 +211,31 @@ export class VizPanel<TOptions = {}, TFieldConfig extends {} = {}> extends Scene
     });
   };
 
+  public getTimeRange = (data?: PanelData) => {
```
Comment (torkelo) [inline packages/scenes/src/components/VizPanel/VizPanel.tsx:214]: should this really be an arrow function?

File: packages/scenes/src/components/VizPanel/VizPanel.tsx [hunk 2]
```diff
@@ -215,6 +211,31 @@ export class VizPanel<TOptions = {}, TFieldConfig extends {} = {}> extends Scene
...
+        sceneTimeRange.getTimeZone(),
+        sceneTimeRange.state.fiscalYearStartMonth,
+        sceneTimeRange.state.UNSAFE_nowDelay
+      );
+    }
+
+    return this.getPanelTimeRange(timeRangeValue, this.getPlugin(), data);
+  }
+
+  private getPanelTimeRange = (timeRange: TimeRange, plugin: PanelPlugin | undefined, data?: PanelData) => {
```
Comment (torkelo) [inline packages/scenes/src/components/VizPanel/VizPanel.tsx:231]: feels a bit confusing to have getTimeRange and getPanelTimeRange functions, do we need both?

File: packages/scenes/src/components/VizPanel/VizPanel.tsx [hunk 3]
```diff
@@ -215,6 +211,31 @@ export class VizPanel<TOptions = {}, TFieldConfig extends {} = {}> extends Scene
     });
   };
 
+  public getTimeRange = (data?: PanelData) => {
+    const liveNowTimer = sceneGraph.findObject(this, (o) => o instanceof LiveNowTimer);
+    const sceneTimeRange = sceneGraph.getTimeRange(this);
+    let timeRangeValue = sceneTimeRange.state.value
+    if (liveNowTimer instanceof LiveNowTimer && liveNowTimer.isEnabled) {
```
Comment (torkelo) [inline packages/scenes/src/components/VizPanel/VizPanel.tsx:218]: why two instanceof checks? (here and on ine 215)

**Review by torkelo — COMMENTED** [review]
Look ok, would be nice to come up with a way where VizPanel would not need to know / check for a liveNowTimer but would require some way to separate time update events that should trigger query and those that should only cause viz-rerender

File: packages/scenes/src/components/VizPanel/VizPanel.tsx (hunk 2 above)
Comment (dprokop) [reply packages/scenes/src/components/VizPanel/VizPanel.tsx:231]: agree, think we can just put `getPanelTimeRange` implementation into the `getTimeRange`, it's like 3 lines of code and the function ain't used anywhere else.

File: packages/scenes/src/components/VizPanel/VizPanel.tsx (hunk 3 above)
Comment (dprokop) [reply packages/scenes/src/components/VizPanel/VizPanel.tsx:218]: @torkelo - `sceneGraph.findObject` ain't typed.

**Review by dprokop — APPROVED** [review]
Looks like a good start. Agree with @torkelo it would be great to make the viz panel independent from the live timer, but I don't think this is critical at the moment. We can address this as a separate improvement to this feature.

File: packages/scenes/src/components/VizPanel/VizPanel.tsx (hunk 3 above)
Comment (torkelo) [reply packages/scenes/src/components/VizPanel/VizPanel.tsx:218]: ah, I see.

**Pull request merged**